File: mobile-flutter/lib/src/features/discovery/domain/movie.dart
class Movie {
  const Movie({
    required this.id,
    required this.title,
    required this.overview,
    this.posterPath = '',
    this.releaseDate = '',
    this.voteAverage = 0,
    this.score,
  });

  final int id;
  final String title;
  final String overview;
  final String posterPath;
  final String releaseDate;
  final double voteAverage;
  final double? score;

  factory Movie.fromJson(Map<String, dynamic> json) {
    return Movie(
      id: (json['id'] as num?)?.toInt() ?? 0,
      title: (json['title'] as String?) ?? 'Untitled',
      overview: (json['overview'] as String?) ?? '',
      posterPath: (json['poster_path'] as String?) ?? '',
      releaseDate: (json['release_date'] as String?) ?? '',
      voteAverage: (json['vote_average'] as num?)?.toDouble() ?? 0,
      score: (json['score'] as num?)?.toDouble(),
    );
  }
}

File: mobile-kotlin/app/src/main/java/com/example/recsystem/data/model/Movie.kt
package com.example.recsystem.data.model

import com.google.gson.annotations.SerializedName

data class Movie(
    val id: Int,
    val title: String,
    val overview: String,
    @SerializedName("poster_path") val posterPath: String = "",
    @SerializedName("release_date") val releaseDate: String = "",
    @SerializedName("vote_average") val voteAverage: Double = 0.0,
    @SerializedName("media_type") val mediaType: String = "movie",
    val score: Double? = null
)
